gsy_framework/community_datasheet/row_converters.py

```diff
@@ -0,0 +1,139 @@
+import re
+import uuid
+from datetime import datetime, timedelta
+from typing import Dict
+
+import pendulum
+
+from gsy_framework.community_datasheet.exceptions import (
+    CommunityDatasheetException, StringToTimedeltaConversionException)
+from gsy_framework.community_datasheet.sheet_headers import (
+    LoadSheetHeader, PVSheetHeader, StorageSheetHeader)
+from gsy_framework.constants_limits import DATE_TIME_FORMAT, ConstSettings
+
+
+class StringToTimedeltaParser:
+    """Class to convert strings to timedelta objects."""
+
+    REGEX = re.compile(r"((?P<hours>\d{1,2}):)?((?P<minutes>\d{1,2}))?")
+
+    @classmethod
+    def parse(cls, time_string: str) -> timedelta:
+        """Parse a string in the format [h]:mm and return a timedelta object.
+
+        Example:
+            >>> StringToTimedeltaParser.parse("24:01")
+            datetime.timedelta(days=1, seconds=60)
+        """
+        parts = cls.REGEX.fullmatch(time_string)
+        if not parts:
+            raise StringToTimedeltaConversionException(
+                f'Value "{time_string}" does not use the supported format. '
+                'Please use [h]:mm (e.g. "1h30m").')
```

Comment:
Does this mean that the hour part is optional?

Reply:
Yes, if you only specify two digits they're interpreted as minutes. E.g.:
```python
from gsy_framework.community_datasheet.row_converters import StringToTimedeltaParser

StringToTimedeltaParser.parse("02")
datetime.timedelta(seconds=120)
```